components/base-node.js

```diff
@@ -65,7 +79,32 @@ module.exports = {
        ar.push(item);
      }
      return ar;
-   } 
+   },
+
+   updatePromise: function(def) {
```

Comment:
This function returns a component factory function. Can we improve the name to better indicate this, such as updaterComponent?